pkg/virt-operator/resource/apply/reconcile.go

```diff
@@ -125,15 +132,51 @@ const (
 )
 
 // Merge all Tolerations, Affinity and NodeSelectos from NodePlacement into pod spec
-func InjectPlacementMetadata(componentConfig *v1.ComponentConfig, podSpec *corev1.PodSpec) {
+func InjectPlacementMetadata(componentConfig *v1.ComponentConfig, podSpec *corev1.PodSpec, nodePlacementOption InfraComponentsNodePlacementOption) {
 	if podSpec == nil {
 		podSpec = &corev1.PodSpec{}
 	}
+
 	if componentConfig == nil || componentConfig.NodePlacement == nil {
-		componentConfig = &v1.ComponentConfig{
-			NodePlacement: &v1.NodePlacement{},
+		if nodePlacementOption == RequireControlPlanePreferNonWorker {
```

Comment:
nit: I always like using switch cases more with enums rather than if-else statement, just a personal preference

Reply:
Yeah I agree :) will change that, thank you!

Reply:
Done.
Please note that I've added a default case that panics if an unknown `nodePlacementOption` is provided. I think it's better to fail fast in this case, and since this value is internal and not user-facing I think it's the right choice here. Any thoughts?

Reply:
I don't think panicking is ever a good idea in production code, can't you just default to a sane behavior in the `default` case?

Reply:
Logging and unit tests are good tools to check that passing unknown values cannot happen, I'd strongly avoid the panic.

Reply:
> I don't think panicking is ever a good idea in production code

I think that's certainly right in most cases. However my concern is that a bug would cause an unexpected behavior that would be hard to spot out. After all this function is completely internal, wouldn't we want to spot the bug right away if an unknown value is being provided? If it was a user-facing function I would speak differently.

> Logging and unit tests are good tools to check that passing unknown values cannot happen

Unit tests? Can you explain how they're helpful here?

w.r.t. logging - Despite easing the debug process a bit, I think the probability of merging a bad PR, causing a surprising behavior, and not being able to spot it for a while is still high.

Reply:
> I think that's certainly right in most cases. However my concern is that a bug would cause an unexpected behavior that would be hard to spot out. After all this function is completely internal, wouldn't we want to spot the bug right away if an unknown value is being provided? If it was a user-facing function I would speak differently.

I think that's one more reason to not panic, panics should be exercised when the program execution cannot continue (fatal error), one good example is segmentation faults, where the program panics because otherwise, it would access non-mapped memory. I don't think this is a good case for panicking, you can just pick a sane default and continue the execution without disruptions.

> Unit tests? Can you explain how they're helpful here?

Fuzzy testing, just pass some random values into the function and test that if the supplied input is invalid it falls back to a default case (which is placement on control plane).

> w.r.t. logging - Despite easing the debug process a bit, I think the probability of merging a bad PR, causing a surprising behavior, and not being able to spot it for a while is still high.

There's no surprising runtime behavior if you fallback to a sane default, actually, you just sanitized input.

It's way more disrupting to panic in this case than just opt for a safe and sane default, I honestly don't see the case for panicking here.

Reply:
I've removed the panic. PTAL.